### PR TITLE
feat(Upload): add transformFile option for transform file before request

### PR DESCRIPTION
**Pull request by lijinke666**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

[https://github.com/react-component/upload/pull/194](https://github.com/react-component/upload/pull/194)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese | 上传之前转换文件          |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/upload/demo/transform-file.md](https://github.com/lijinke666/ant-design/blob/feature/components/upload/demo/transform-file.md)
[View rendered components/upload/index.en-US.md](https://github.com/lijinke666/ant-design/blob/feature/components/upload/index.en-US.md)
[View rendered components/upload/index.zh-CN.md](https://github.com/lijinke666/ant-design/blob/feature/components/upload/index.zh-CN.md)

### CODE REVIEW

**Comment by lijinke666** [comment]
@zombieJ 老哥 麻烦看下噻

File: components/upload/index.en-US.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Uploading is the process of publishing information (web pages, text, pictures, v
 | onChange | A callback function, can be executed when uploading state is changing, see [onChange](#onChange) | Function | - |  |
 | onPreview | A callback function, will be executed when file link or preview icon is clicked | Function(file) | - |  |
 | onRemove | A callback function, will be executed when removing file button is clicked, remove event will be prevented when return value is `false` or a Promise which resolve(false) or reject | Function(file): `boolean | Promise` | - |  |
+| transformFile   | Customize transform file before request | Function(file): `any | Promise<any>` | - |  |
```
Comment (zombieJ) [inline components/upload/index.en-US.md]: 添加一下版本信息，3.21.0

File: components/upload/interface.tsx [hunk 2]
```diff
@@ -82,6 +82,7 @@ export interface UploadProps {
   locale?: UploadLocale;
   id?: string;
   previewFile?: PreviewFileHandler;
+  transformFile?: (file: UploadFile) => any | Promise<any>;
```
Comment (zombieJ) [inline components/upload/interface.tsx]: `Blob | File`?

Comment (lijinke666) [reply components/upload/interface.tsx]: 这里我想的是 他可能转换 null  string(base64) File Blob  用 any 合适一点

Comment (zombieJ) [reply components/upload/interface.tsx]: 那可以加个 string 类型，不用 any

**Comment by lijinke666** [comment]
@zombieJ 可以上车了不

**Review by zombieJ — APPROVED** [review]

**Pull request merged**